homeassistant/components/zwave_me/__init__.py

```diff
@@ -0,0 +1,126 @@
+"""The Z-Wave-Me WS integration."""
+import asyncio
+import logging
+
+from zwave_me_ws import ZWaveMe, ZWaveMeData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_connect, dispatcher_send
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, PLATFORMS, ZWAVE_PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def get_uuid(url, token):
+    """Get an uuid from Z-Wave-Me."""
+    conn = ZWaveMe(url=url, token=token)
+    uuid = None
+    if await conn.get_connection():
+        uuid = await conn.get_uuid()
+    await conn.close_ws()
+    return uuid
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Z-Wave-Me from a config entry."""
+    hass.data[DOMAIN] = {}
+    controller = hass.data[DOMAIN][entry.entry_id] = ZWaveMeController(hass, entry)
+    if await controller.async_establish_connection():
+        await async_setup_platforms(hass, entry, controller)
+        return True
+    raise ConfigEntryNotReady()
+
+
+async def async_unload_entry(hass, entry):
+    """Unload a config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+class ZWaveMeController:
+    """Main ZWave-Me API class."""
+
+    def __init__(self, hass: HomeAssistant, config: ConfigEntry) -> None:
+        """Create the API instance."""
+        self.device_ids: set = set()
+        self._hass = hass
+        self.config = config
+        self.zwave_api = ZWaveMe(
+            on_device_create=self.on_device_create,
+            on_device_update=self.on_device_update,
+            on_new_device=self.add_device,
+            token=self.config.data["token"],
+            url=self.config.data["url"],
+            platforms=ZWAVE_PLATFORMS,
+        )
+        self.platforms_inited = False
+
+    async def async_establish_connection(self):
+        """Get connection status."""
+        is_connected = await self.zwave_api.get_connection()
+        if is_connected:
+            self.platforms_inited = True
```

Comment:
The platforms are ready when they have been gathered in `async_setup_platforms`. I'd set this attribute there.

Reply:
> Please take care not to run blocking calls inside the event loop. See here eg: https://github.com/lawfulchaos/ZWave-Me-WS/blob/48034de8cc8e8d9ef3dda5eee1d0b4a53d3943ab/zwave_me_ws/ZWaveMe.py#L51
> 
> If you want to use asyncio in the library I'd suggest changing the websocket library to a library that supports asyncio natively, so you don't need to worry about mixing sync and async code.

We prefer not to make this change right now. It will require one more full rework of the WS library. Please accept this PR without this particular change.

In addition please let us know what performance consequences do to get here? I've noticed that most WS-based integrations don't care about async so I would assume this is not a must-have. We can either plan this for future rework or keep it as is.